library/Requests/Transport/cURL.php

```diff
@@ -546,4 +561,25 @@ public static function test($capabilities = array()) {
 
 		return true;
 	}
+
+	/**
+	 * Get the correct "Expect" header for the given request data.
+	 *
+	 * @param string|array $data Data to send either as the POST body, or as parameters in the URL for a GET/HEAD.
+	 * @return string The "Expect" header.
+	 */
+	protected function get_expect_header($data) {
+		if (is_array($data)) {
+			$iterator = new RecursiveIteratorIterator(new RecursiveArrayIterator($data));
+			$bytesize = 0;
+
+			foreach ($iterator as $datum) {
+				$bytesize += strlen((string) $datum);
```

Comment:
I think it might be worth bailing out here immediately once the `bytesize` limit is reached.